homeassistant/components/zha/core/helpers.py

```diff
@@ -157,3 +162,42 @@ def get_attr_id_by_name(cluster, attr_name):
     """Get the attribute id for a cluster attribute by its name."""
     return next((attrid for attrid, (attrname, datatype) in
                  cluster.attributes.items() if attr_name == attrname), None)
+
+
+async def get_matched_clusters(source_zha_device, target_zha_device):
+    """Get matched input/output cluster pairs for 2 devices."""
+    source_clusters = source_zha_device.async_get_zha_clusters()
+    target_clusters = target_zha_device.async_get_zha_clusters()
+    clusters_to_bind = []
+
+    for endpoint_id in source_clusters:
+        for cluster_id in source_clusters[endpoint_id][OUT]:
+            if cluster_id in BINDABLE_CLUSTERS:
+                for t_endpoint_id in target_clusters:
+                    if cluster_id in target_clusters[t_endpoint_id][IN]:
+                        cluster_pair = ClusterPair(
+                            source_cluster=source_clusters[
+                                endpoint_id][OUT][cluster_id],
+                            target_cluster=target_clusters[
+                                t_endpoint_id][IN][cluster_id]
+                        )
+                        clusters_to_bind.append(cluster_pair)
+    return clusters_to_bind
+
+
+@callback
+def async_is_bindable_target(source_zha_device, target_zha_device):
+    """Determine if target is bindable to source."""
+    source_clusters = source_zha_device.async_get_zha_clusters()
+    target_clusters = target_zha_device.async_get_zha_clusters()
+
+    for endpoint_id in source_clusters:
+        for t_endpoint_id in target_clusters:
+            matches = set(
+                source_clusters[endpoint_id][OUT].keys()
+                ).intersection(
+                    target_clusters[t_endpoint_id][IN].keys()
+                )
+            if bool(set(matches) & set(BINDABLE_CLUSTERS)):
```

Comment:
`matches` is already a set ? 

Also convert `BINDABLE_CLUSTERS` only once to a set.